esi/src/config.rs

```diff
@@ -3,25 +3,18 @@
 /// ## Usage Example
 /// ```rust,no_run
 /// let config = esi::Configuration::default()
-///     .with_namespace("app")
-///     .with_recursion();
-///
-/// let processor = esi::Processor::new(config);
+///     .with_namespace("app");
 /// ```
 #[derive(Clone, Debug)]
 pub struct Configuration {
     /// The XML namespace to use when scanning for ESI tags. Defaults to `esi`.
     pub namespace: String,
```

Comment:
I don't know if this is worth changing, but it bothers me when XML consumers rely on a specific name for a namespace instead of processing the `xmlns` attributes and looking for the right namespace URI. In principle, users should never need to configure the namespace.

I see `quick_xml` has an [NsReader](https://docs.rs/quick-xml/latest/quick_xml/struct.NsReader.html) helper for doing namespace resolution, so maybe this is easy to fix?